### PR TITLE
add capability for the pipeline to create/destroy pktin netlink sockets

### PR DESCRIPTION
**Pull request by harshsin**
Reviewer: @rlane 

Commit Summary: Allow pipelines to register netlink sockets to receive pktin's. Pipeline will set the callback to be associated with each socket which will be invoked when a packet is recv'd on that socket. 
Pipeline will also set the ratelimiter associated with that socket.

Todo: Add timer to pause/resume the socket on ratelimiter overflow.

### CODE REVIEW

**Comment by rlane** [comment]
Merge conflict.

**Comment by rlane** [comment]
Go ahead and convert pipeline_standard to the new API so we can see an example of how it will be used.

**Comment by harshsin** [comment]
Converted pipeline_standard to new socket registration mechanism. Please take a look.

**Comment by rlane** [comment]
Finished first round of review.

**Comment by harshsin** [comment]
Thanks for taking a look. Pushed another commit addressing your comments.

File: modules/OVSDriver/module/src/ovs_driver.c [hunk 1]
```diff
@@ -51,6 +51,8 @@ bool ind_ovs_disable_megaflows = false;
 uint32_t ind_ovs_salt;
 uint16_t ind_ovs_inband_vlan = VLAN_INVALID;
 
+struct ind_ovs_pktin_socket pktout_soc;
```
Comment (rlane) [inline modules/OVSDriver/module/src/ovs_driver.c]: Add a comment saying what this variable is used for.

**Comment by rlane** [comment]
Looks good. Feel free to self-ack.

**Comment by harshsin** [comment]
Pushed a commit with the changes.

SELF-ACK

**Comment by bsn-abat** [comment]
ABAT: START_MERGE:  Log file at http://trusty1/abat/2015.03.13.2231-m.ivs.master/abat.log

**Comment by bsn-abat** [comment]
ABAT: ACCEPT:  Successfully merged

In case you want to see the build log, check out :
Log file at http://trusty1/abat/2015.03.13.2231-m.ivs.master/abat.log
/cc @rlane

**Pull request merged**